cmd/amazon-cloudwatch-agent-target-allocator/config/config.go

```diff
@@ -207,26 +209,34 @@ func Load() (*Config, string, error) {
 
 // ValidateConfig validates the cli and file configs together.
 func ValidateConfig(config *Config) error {
-	scrapeConfigsPresent := (config.PromConfig != nil && len(config.PromConfig.ScrapeConfigs) > 0)
+	scrapeConfigsPresent := config.PromConfig != nil && len(config.PromConfig.ScrapeConfigs) > 0
 	if !(config.PrometheusCR.Enabled || scrapeConfigsPresent) {
 		return fmt.Errorf("at least one scrape config must be defined, or Prometheus CR watching must be enabled")
 	}
 	return nil
 }
 
-func (c HTTPSServerConfig) NewTLSConfig() (*tls.Config, error) {
-	cert, err := tls.LoadX509KeyPair(c.TLSCertFilePath, c.TLSKeyFilePath)
-	if err != nil {
-		return nil, err
-	}
+func (c HTTPSServerConfig) NewTLSConfig(ctx context.Context) (*tls.Config, error) {
 	tlsConfig := &tls.Config{
-		Certificates: []tls.Certificate{cert},
-		ClientAuth:   tls.NoClientCert,
-		MinVersion:   tls.VersionTLS12,
+		MinVersion: tls.VersionTLS12,
```

Comment:
even though it is not directly related to this PR , can we enforce TLS 1.3?

Reply:
yeah i can update that